language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/EnumMemberNodeContext.java

```diff
@@ -32,13 +43,42 @@
  */
 @JavaSPIService("org.ballerinalang.langserver.commons.completion.spi.BallerinaCompletionProvider")
 public class EnumMemberNodeContext extends AbstractCompletionProvider<EnumMemberNode> {
+
     public EnumMemberNodeContext() {
         super(EnumMemberNode.class);
     }
 
     @Override
     public List<LSCompletionItem> getCompletions(BallerinaCompletionContext ctx, EnumMemberNode node)
             throws LSCompletionException {
-        return Collections.emptyList();
+        List<LSCompletionItem> completionItems = new ArrayList<>();
+        NonTerminalNode nodeAtCursor = ctx.getNodeAtCursor();
+        List<Symbol> visibleSymbols = new ArrayList<>();
+        if (!inEnumMemberValueContext(ctx, node)) {
+            return completionItems;
+        }
+        if (QNameReferenceUtil.onQualifiedNameIdentifier(ctx, nodeAtCursor)) {
+            QualifiedNameReferenceNode nameRef = (QualifiedNameReferenceNode) nodeAtCursor;
+            Predicate<Symbol> filter = symbol -> symbol.kind() == SymbolKind.CONSTANT
```

Comment:
You can reuse this filter in both cases. Shall we refactor?

Reply:
done!